docs/ui/components/MarkdownActions/MarkdownActionsDropdown.tsx

```diff
@@ -0,0 +1,122 @@
+import { useRouter } from 'next/compat/router';
+import { useCallback, useMemo } from 'react';
+
+import * as Dropdown from '~/ui/components/Dropdown';
+import { githubRawUrl, getPageMdxFilePath } from '~/ui/components/Footer/utils';
+import { prepareMarkdownForCopyAsync } from '~/ui/components/MarkdownActions/processMarkdown';
+import { FOOTNOTE } from '~/ui/components/Text';
+
+const getPrompt = (markdownUrl: string) =>
+  encodeURIComponent(`Read from ${markdownUrl} so I can ask questions about it.`);
+
+export function MarkdownActionsDropdown() {
+  const router = useRouter();
+
+  const pathname = router?.pathname;
+  const asPath = router?.asPath;
+
+  const rawMarkdownUrl = useMemo(() => {
+    if (!pathname) {
+      return null;
+    }
+
+    const filePath = getPageMdxFilePath(pathname);
+    if (!filePath) {
+      return null;
+    }
+
+    return githubRawUrl(pathname);
+  }, [pathname]);
+
+  const handleCopyMarkdown = useCallback(async () => {
+    if (!rawMarkdownUrl) {
+      return;
+    }
+
+    try {
+      const response = await fetch(rawMarkdownUrl);
+      if (!response.ok) {
+        throw new Error(`Failed to fetch markdown: ${response.status}`);
+      }
+
+      const markdown = await response.text();
+
+      if (!navigator.clipboard?.writeText) {
+        throw new Error('Clipboard API unavailable');
+      }
+
+      const preparedMarkdown = await prepareMarkdownForCopyAsync(markdown, {
+        path: asPath ?? pathname ?? '',
+      });
+      await navigator.clipboard.writeText(preparedMarkdown);
+    } catch (error) {
+      console.error('Unable to copy markdown content', error);
+    }
+  }, [rawMarkdownUrl, asPath, pathname]);
+
+  const chatGptUrl = useMemo(() => {
+    if (!rawMarkdownUrl) {
+      return null;
+    }
+    return `https://chat.openai.com/?q=${getPrompt(rawMarkdownUrl)}`;
+  }, [rawMarkdownUrl]);
+
+  const claudeUrl = useMemo(() => {
+    if (!rawMarkdownUrl) {
+      return null;
+    }
+    return `https://claude.ai/new?q=${getPrompt(rawMarkdownUrl)}`;
+  }, [rawMarkdownUrl]);
+
+  const dropdownTrigger = (
+    <Button
+      theme="quaternary"
+      className="justify-center pl-2.5 pr-2"
+      aria-haspopup="menu"
+      aria-label="Copy Markdown and AI actions">
+      <div className="flex flex-row items-center gap-1.5">
+        <FOOTNOTE crawlable={false} theme="secondary" className="whitespace-nowrap">
+          Copy
+        </FOOTNOTE>
+        <ChevronDownIcon className="icon-xs text-icon-secondary" />
+      </div>
+    </Button>
+  );
+
+  return (
+    <Dropdown.Dropdown trigger={<div>{dropdownTrigger}</div>} sideOffset={8}>
+      <Dropdown.Item
+        label="Copy Markdown"
+        Icon={Copy04Icon}
+        onSelect={handleCopyMarkdown}
+        disabled={!rawMarkdownUrl}
+      />
+      <Dropdown.Item
+        label="View Markdown"
+        Icon={EyeIcon}
+        href={rawMarkdownUrl ?? undefined}
+        openInNewTab
+        disabled={!rawMarkdownUrl}
```

Comment:
For all three options, I think it's better not to render them when there's no URL.

Reply:
Not sure I understand. Can you elaborate?